### PR TITLE
clustermesh: add support for max-connected-clusters

### PR DESCRIPTION
**Pull request by thorn3r**
The flag 'max-connected-clusters' is a cilium-agent option that allows the user to extend the number of clusters that can be connected in a Clustermesh. This changes the maximum ClusterID that can be used, and thus cilium-cli will now validate according to the agent's configuration.

Additionally, all clusters in a Clustermesh must be configured with the same 'max-connected-clusters', and this is now validated on connection.

Associated PR introducing `max-connected-clusters`  option: https://github.com/cilium/cilium/pull/27520

### CODE REVIEW

**Review by jrajahalme — CHANGES_REQUESTED** [review]
Cluster ID is used in 8-bit datapath fields, so it can not be larger than 255. So at the minimum some validation needs to be added to check that this maximum is not surpassed.

**Review by derailed — APPROVED** [review]
@thorn3r Nice work! LGTM.

**Comment by thorn3r** [comment]
thanks @derailed !

@jrajahalme I forgot to link the associated cilium/cilium PR: https://github.com/cilium/cilium/pull/27520
TLDR: the 8-bit field can be configured to 9-bit via the new flag. This cilium-cli PR is to support configuring this flag and allow validation for `clustermesh connect` for clusters using a non-default configuration.

**Comment by thorn3r** [comment]
converting this and the associated [Cilium PR](https://github.com/cilium/cilium/pull/27520) back into a draft for now until https://github.com/cilium/cilium/pull/27823 is merged. I don't anticipate any changes to the flag itself, but will hold for now

**Comment by thorn3r** [comment]
putting this back into review alongside https://github.com/cilium/cilium/pull/27520

**Review by jrajahalme — APPROVED** [review]

**Comment by squeed** [comment]
@thorn3r needs a rebase, buddy :-(
